Python/chapter01/1.8 - Zero Matrix/miguel_1.8_sol.py

```diff
@@ -0,0 +1,135 @@
+"""
+Python version 3.7.0
+1.8 - Zero Matrix
+Write an algorithm such that if an element in an MxN matrix is 0, its entire row and column are set to 0
+"""
+import unittest
+from typing import List
+
+
+def zero_matrix(matrix: List[List[int]]) -> List[List[int]]:
+    """
+    zero_matrix will take in an MxN matrix and when a 0 element is found, the whole column and row
+    will be set to 0.
+    Runtime: O(M * N)
+    Space Complexity: O(min(M, N))
```

Comment:
can you elaborate why this is `min(M, N)`?

Reply:
Initially I thought the worst case for storing zeroed columns and rows was when there were 0s along the diagonal (The diagonal count of an MxN matrix is the smaller value between M and N, hence min(M, N)).  However, I was wrong.
I overlooked the case when all values along a row or column are 0.
Solution and test cases are updated accordingly to account for that.  My new solution takes a different approach.